### PR TITLE
http2: Update nghttp2 to v1.40.0

### PR DESCRIPTION
**Pull request by yanavlasov**
Set nghttp2 internal flood mitigation threshold back to 10K to avoid any changes in Envoy's codec behavior.

Risk Level:
Testing:
Docs Changes:
Release Notes:

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -118,9 +118,9 @@ REPOSITORY_LOCATIONS = dict(
         urls = ["https://github.com/LuaJIT/LuaJIT/archive/v2.1.0-beta3.tar.gz"],
     ),
     com_github_nghttp2_nghttp2 = dict(
-        sha256 = "25b623cd04dc6a863ca3b34ed6247844effe1aa5458229590b3f56a6d53cd692",
-        strip_prefix = "nghttp2-1.39.1",
-        urls = ["https://github.com/nghttp2/nghttp2/releases/download/v1.39.1/nghttp2-1.39.1.tar.gz"],
+        sha256 = "fc820a305e2f410fade1a3260f09229f15c0494fc089b0100312cd64a33a38c0",
+        strip_prefix = "nghttp2-1.39.2",
+        urls = ["https://github.com/nghttp2/nghttp2/releases/download/v1.39.2/nghttp2-1.39.2.tar.gz"],
```
Comment (moderation) [inline bazel/repository_locations.bzl]: I would suggest jumping to the latest version 1.40.0. This version is compiling and working for me in local testing.
```
        sha256 = "eb9d9046495a49dd40c7ef5d6c9907b51e5a6b320ea6e2add11eb8b52c982c47",
        strip_prefix = "nghttp2-1.40.0",
        urls = ["https://github.com/nghttp2/nghttp2/releases/download/v1.40.0/nghttp2-1.40.0.tar.gz"],
```

Comment (yanavlasov) [reply bazel/repository_locations.bzl]: It does not work for me due to this https://github.com/nghttp2/nghttp2/commit/28b1f0b90f82cccb3f1977ed3b580d2c98421f94 the name of the nghttp2 static library has changed. Did you make some other changes to the build system to build 1.40.0?

Comment (moderation) [reply bazel/repository_locations.bzl]: @yanavlasov I can't see your comment for some reason but there was a single change I made for 1.40.0 after working out the change in the commit log.
```diff
diff --git a/bazel/foreign_cc/BUILD b/bazel/foreign_cc/BUILD
index 82994cbd4..3c15463f3 100644
--- a/bazel/foreign_cc/BUILD
+++ b/bazel/foreign_cc/BUILD
@@ -156,7 +156,7 @@ envoy_cmake_external(
     lib_source = "@com_github_nghttp2_nghttp2//:all",
     static_libraries = select({
         "//bazel:windows_x86_64": ["nghttp2.lib"],
-        "//conditions:default": ["libnghttp2.a"],
+        "//conditions:default": ["libnghttp2_static.a"],
     }),
 )
```

Comment (moderation) [reply bazel/repository_locations.bzl]: The PR that led me to this change is https://github.com/nghttp2/nghttp2/pull/1394

Comment (yanavlasov) [reply bazel/repository_locations.bzl]: Sorry, I've deleted my comment as I figured out where expected output is specified.

**Pull request merged**